### PR TITLE
docs(sylvia-derive): Document inner types

### PR DESCRIPTION
**Pull request by jawoznia**

### CODE REVIEW

File: sylvia-derive/src/contract.rs [hunk 1]
```diff
@@ -52,6 +67,7 @@ impl<'a> ContractInput<'a> {
         }
     }
 
+    /// Process the input and generate the contract code.
```
Comment (kulikthebird) [inline sylvia-derive/src/contract.rs:70]: Typo: `Processes` and `generates`

File: sylvia-derive/src/contract/communication/executor.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ use crate::types::msg_field::MsgField;
 use crate::types::msg_variant::{MsgVariant, MsgVariants};
 use crate::utils::SvCasing;
 
+/// Emits execute helper
```
Comment (kulikthebird) [inline sylvia-derive/src/contract/communication/executor.rs:13]: Here we can also provide some reference to what is the `Executor`

File: sylvia-derive/src/contract/communication/querier.rs [hunk 3]
```diff
@@ -10,6 +10,7 @@ use crate::types::msg_field::MsgField;
 use crate::types::msg_variant::{MsgVariant, MsgVariants};
 use crate::utils::SvCasing;
 
+/// Emits query helper
```
Comment (kulikthebird) [inline sylvia-derive/src/contract/communication/querier.rs:13]: Same as in Executor comment

File: sylvia-derive/src/contract/communication/wrapper_msg.rs [hunk 4]
```diff
@@ -9,7 +9,8 @@ use syn::fold::Fold;
 use syn::spanned::Spanned;
 use syn::{Ident, ItemImpl, Type};
 
-/// Glue message is the message composing Exec/Query messages from several traits
+/// Glue message is the message composing Exec/Query/Sudo messages from several traits and a
```
Comment (kulikthebird) [inline sylvia-derive/src/contract/communication/wrapper_msg.rs:12]: Adding a reference or more details here will be beneficial for newcomers. For e.g. some reasoning why sylvia generates additional wrapper messages, why it should be parsed in a different way etc.

File: sylvia-derive/src/contract/mt.rs [hunk 5]
```diff
@@ -22,6 +22,7 @@ fn get_ident_from_type(contract_name: &Type) -> &Ident {
     &segment.ident
 }
 
+/// Emits helpers for testing contract messages using MultiTest.
```
Comment (kulikthebird) [inline sylvia-derive/src/contract/mt.rs:25]: What kind of helpers?

File: sylvia-derive/src/entry_points.rs [hunk 6]
```diff
@@ -42,13 +47,15 @@ impl<'a> EntryPointInput<'a> {
         Self { item, args }
     }
 
+    /// Process the input and generate the interface code.
```
Comment (kulikthebird) [inline sylvia-derive/src/entry_points.rs:50]: We could enrich this description with information about exporting several functions related to entrypoints

File: sylvia-derive/src/interface.rs [hunk 7]
```diff
@@ -17,7 +17,17 @@ use crate::types::msg_variant::MsgVariants;
...
+///
+/// Generates `sv` module containing:
+///     - [Messages](https://cosmwasm-docs.vercel.app/sylvia/macros/generated-types/message-types#interface-messages)
+///         - ExecMsg
+///         - QueryMsg
+///         - SudoMsg
+///     - [MultiTest](https://cosmwasm-docs.vercel.app/sylvia/macros/generated-types/multitest#proxy-trait) helpers
+///     - [Querier](https://cosmwasm-docs.vercel.app/cw-multi-test) trait implementation
+///     - [Executor](https://cosmwasm-docs.vercel.app/cw-multi-test) trait implementation
+///     - Api trait implementation
```
Comment (kulikthebird) [inline sylvia-derive/src/interface.rs:30]: Let's add a reference here

File: sylvia-derive/src/contract.rs [hunk 8]
```diff
@@ -21,11 +21,26 @@ use crate::types::msg_variant::MsgVariants;
...
+///         - QueryMsg
+///         - SudoMsg
+///         - MigrateMsg
+///         - ContractExecMsg
+///         - ContractQueryMsg
+///         - ContractSudoMsg
+///     - [MultiTest](https://cosmwasm-docs.vercel.app/sylvia/macros/generated-types/multitest) helpers
+///     - [Querier](https://cosmwasm-docs.vercel.app/cw-multi-test) trait implementation
+///     - [Executor](https://cosmwasm-docs.vercel.app/cw-multi-test) trait implementation
+///     - Api trait implementation
```
Comment (kulikthebird) [inline sylvia-derive/src/contract.rs:39]: Let's add a reference here

File: sylvia-derive/src/interface.rs [hunk 9]
```diff
@@ -81,6 +91,7 @@ impl<'a> InterfaceInput<'a> {
         }
     }
 
+    /// Process the input and generate the interface code.
```
Comment (kulikthebird) [inline sylvia-derive/src/interface.rs:94]: Typo: `Processes` and `generates`

File: sylvia-derive/src/interface/communication/api.rs [hunk 10]
```diff
@@ -6,6 +6,7 @@ use proc_macro2::TokenStream;
 use quote::quote;
 use syn::ItemTrait;
 
+/// Emits `InterfaceMessagesApi` trait.
```
Comment (kulikthebird) [inline sylvia-derive/src/interface/communication/api.rs:9]: Let's add some more details on what is `InterfaceMessagesApi`

File: sylvia-derive/src/interface/communication/querier.rs [hunk 11]
```diff
@@ -11,6 +11,7 @@ use crate::types::msg_field::MsgField;
 use crate::types::msg_variant::{MsgVariant, MsgVariants};
 use crate::utils::SvCasing;
 
+/// Emits query helper
```
Comment (kulikthebird) [inline sylvia-derive/src/interface/communication/querier.rs:14]: Same as in `Executor` helpers

File: sylvia-derive/src/types/associated_types.rs [hunk 12]
```diff
@@ -69,8 +76,28 @@ impl<'a> AssociatedTypes<'a> {
...
+    /// Returns `None` if it is not.
+    pub fn emit_contract_custom_type_accessor(
+        &self,
+        trait_name: &Ident,
+        type_name: &str,
+    ) -> Option<Type> {
+        match self
+            .as_names()
+            .find(|name| name.to_string().as_str() == type_name)
+        {
```
Comment (kulikthebird) [inline sylvia-derive/src/types/associated_types.rs:90]:
```rust
.map(|name| => {
    let type_name = Ident::new(type_name, name.span());
    parse_quote! { <ContractT as #trait_name>:: #type_name }
})
```

**Review by kulikthebird — CHANGES_REQUESTED** [review]
It's a move towards a good direction, definitely Sylvia deserves an internal documentation that will demystify its machinery. I left some comments to the code and docstrings where I believe we could add some more details. Overall I like this change!

File: sylvia-derive/src/interface.rs (hunk 7 above)
Comment (jawoznia) [reply sylvia-derive/src/interface.rs:30]: It's currently missing. Will add that in the future

File: sylvia-derive/src/contract.rs (hunk 8 above)
Comment (jawoznia) [reply sylvia-derive/src/contract.rs:39]: same

File: sylvia-derive/src/types/associated_types.rs (hunk 12 above)
Comment (jawoznia) [reply sylvia-derive/src/types/associated_types.rs:90]: good call

**Review by kulikthebird — APPROVED** [review]

**Pull request merged**